tests/unit/tuner/callback/test_model_checkpoint.py

```diff
@@ -0,0 +1,177 @@
+import pytest
+import tensorflow as tf
+import torch
+import paddle
+from finetuner.tuner.callback import ModelCheckpointCallback
+from finetuner.tuner.base import BaseTuner
+import finetuner
+from finetuner.toydata import generate_fashion
+import os
+import tempfile
+import numpy as np
+
+
+@pytest.fixture(scope="module")
+def keras_model() -> BaseTuner:
+    embed_model = tf.keras.Sequential(
+        [
+            tf.keras.layers.Flatten(input_shape=(28, 28)),
+            tf.keras.layers.Dense(128, activation='relu'),
+            tf.keras.layers.Dense(32),
+        ]
+    )
+    return embed_model
+
+
+@pytest.fixture(scope="module")
+def pytorch_model() -> BaseTuner:
+    embed_model = torch.nn.Sequential(
+        torch.nn.Flatten(),
+        torch.nn.Linear(
+            in_features=28 * 28,
+            out_features=128,
+        ),
+        torch.nn.ReLU(),
+        torch.nn.Linear(in_features=128, out_features=32),
+    )
+    return embed_model
+
+
+@pytest.fixture(scope="module")
+def paddle_model() -> BaseTuner:
+    embed_model = paddle.nn.Sequential(
+        paddle.nn.Flatten(),
+        paddle.nn.Linear(
+            in_features=28 * 28,
+            out_features=128,
+        ),
+        paddle.nn.ReLU(),
+        paddle.nn.Linear(in_features=128, out_features=32),
+    )
+    return embed_model
+
+
+def test_keras_model(keras_model: BaseTuner):
+    with tempfile.TemporaryDirectory() as tmpdirname:
```

Comment:
You can use the `tmpdir` fixture. See for example [here](https://github.com/jina-ai/jina/blob/898d15731e2f1faf414b4641d9e438b0bef8f324/tests/unit/flow-construct/test_flow_visualization.py#L13)